carrier.go

```diff
@@ -50,34 +48,58 @@ type CarrierType struct {
 }
 
 type carrierAccountRequest struct {
-	CarrierAccount *CarrierAccount `json:"carrier_account,omitempty"`
+	Data *CarrierAccount `json:"carrier_account,omitempty"`
 }
 
-func (c *Client) selectCarrierAccountCreationEndpoint(account CarrierAccount) string {
+// UpsCarrierAccountCreationParameters contains the parameters needed to create a new UPS carrier account.
+type UpsCarrierAccountCreationParameters struct {
+	Type          string `json:"type,omitempty"`
+	Description   string `json:"description,omitempty"`
+	Reference     string `json:"reference,omitempty"`
+	AccountNumber string `json:"account_number,omitempty"`
+}
+
+type upsCarrierAccountCreationRequest struct {
+	Data *UpsCarrierAccountCreationParameters `json:"ups_oauth_registrations,omitempty"`
+}
+
+// UpsCarrierAccountUpdateParameters contains the parameters needed to update a UPS carrier account.
+type UpsCarrierAccountUpdateParameters struct {
+	AccountNumber string `json:"account_number,omitempty"`
+}
```

Comment:
Nit: add TODO comments here to allow `description` and `reference` when the endpoint allows it

Reply:
I'd like us to avoid polluting external code with internal concerns. We need to address this internally with the API team to ensure it's even ever going to happen instead of alluding to it to users before we know.